sdk/java/src/main/java/sawtooth/sdk/client/State.java

```diff
@@ -77,6 +79,10 @@ public State(Stream stream, String contextId) {
     }
     Map<String, ByteString> results = new HashMap<String, ByteString>();
     if (getResponse != null) {
+      if (getResponse.getStatus() == TpStateGetResponse.Status.AUTHORIZATION_ERROR) {
+        throw new InvalidTransactionException(
+          "Tried to get unauthorized address" + addresses.toString()) ;
```

Comment:
A nit, but you will probably want a space after "...address" to make string readable.  Same for line 135 below.